File: system-test/java/src/test/java/com/mycompany/myshop/systemtest/legacy/mod05/smoke/system/MyShopBaseSmokeTest.java
package com.mycompany.myshop.systemtest.legacy.mod05.smoke.system;

import com.mycompany.myshop.systemtest.legacy.mod05.base.BaseDriverTest;
import com.mycompany.myshop.testkit.driver.port.dtos.GoToMyShopRequest;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static com.mycompany.myshop.testkit.common.ResultAssert.assertThatResult;

public abstract class MyShopBaseSmokeTest extends BaseDriverTest {
    @BeforeEach
    void setUp() {
        setMyShopDriver();
    }

    protected abstract void setMyShopDriver();

    @Test
    void shouldBeAbleToGoToMyShop() {
        var result = myShopDriver.goToMyShop(GoToMyShopRequest.builder().build());
        assertThatResult(result).isSuccess();
    }
}

File: system-test/dotnet/SystemTests/Legacy/Mod05/SmokeTests/System/MyShopBaseSmokeTest.cs
using Common;
using Driver.Port.Dtos;
using SystemTests.Legacy.Mod05.Base;
using Shouldly;
using Xunit;

namespace SystemTests.Legacy.Mod05.SmokeTests.System;

public abstract class MyShopBaseSmokeTest : BaseDriverTest
{
    protected abstract Task SetUpMyShopDriverAsync();

    public override async Task InitializeAsync()
    {
        await base.InitializeAsync();
        await SetUpMyShopDriverAsync();
    }

    [Fact]
    public async Task ShouldBeAbleToGoToMyShop()
    {
        var result = await _shopDriver!.GoToMyShopAsync(new GoToMyShopRequest());
        result.ShouldBeSuccess();
    }
}










